src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -92,7 +112,15 @@ public CalculationResults GetSuggestedExpensesOffers(IEntryList entryList, IGoal
             {
                 monthlySuggestions.Add(new SavingSuggestionByImportance(averagesOfAmountByImportanceAdjusted[i - 1], averagesOfAmountByImportanceDefault[i - 1], (Importance)i));
             }
-            return new CalculationResults(entrySuggestions, monthlySuggestions, timesToRepeatSaving);
+            if (!generateRandomData)
+            {
+                return new CalculationResults(entrySuggestions, monthlySuggestions, timesToRepeatSaving);
+            }
+            else
+            {
+                return new CalculationResults(entrySuggestions, monthlySuggestions, timesToRepeatSaving * -1);
+            }
```

Comment:
This if else also can be shortened:
       
     return !generateRandomData ? new CalculationResults(entrySuggestions, monthlySuggestions,mesToRepeatSaving) 
                : new CalculationResults(entrySuggestions, monthlySuggestions, timesToRepeatSaving * -1);